### DISCUSSION TITLE
Test failures with test suite in void-packages's CI

### DISCUSSION THREAD

**Discussion by triallax** [opensteno/plover#1587] [Technical Discussion]
Hello, I'm working to package Plover for Void Linux's repos (https://github.com/void-linux/void-packages/pull/42077). For some reason though, some tests are failing in the CI, even though they succeed locally. See https://github.com/void-linux/void-packages/actions/runs/4097979950/jobs/7066752808#step:7:766 for the failures. Any idea what might be happening here and how to diagnose and solve the issue?

**Comment by user202729** [comment]
Basically if the file is read-only in the file system (which is ensured by the line `dict_path.chmod(0o440)` above) then Plover should be able to detect it and mark it as read-only (shown with a padlock to the left of the dictionary in the GUI)

For some reason the read-only status is not detected (it's connected with os.access, refer to
https://github.com/openstenoproject/plover/blob/3066a9a47269861ac1d66f8f05cdb26f7251b98d/plover/steno_dictionary.py#L66
), or the chmod() line is not working. Maybe the file system is not the usual one?

**Reply by triallax** [reply]
Apparently the CI runs the tests as root, do you think that might be the culprit?

**Reply by user202729** [reply]
Looks correct.

https://unix.stackexchange.com/q/412234

Run as root:

```python
Python 3.10.8 (main, Nov  1 2022, 14:18:21) [GCC 12.2.0] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> with open("/tmp/a.txt", "w") as f: pass
... 
>>> import os
>>> os.chmod("/tmp/a.txt", 0)
>>> os.access("/tmp/a.txt", os.W_OK)
True  # for a normal user this would be "False"
>>> with open("/tmp/a.txt", "w") as f: pass
... 
>>> 
```

**Reply by triallax** [reply]
Thanks, that makes sense, I'll try to work around it on Void's side (for now I'll probably simply skip those tests, but longer-term it's probably a better idea to simply not run them as root).

**Comment by triallax** [comment]
I'm glad to say that my PR has been merged, and Plover is now available in the Void Linux repositories as `plover`! I also added a section in [the installation section of the wiki](https://github.com/openstenoproject/plover/wiki/Installation-Guide#install-plover-on-void-linux) documenting this.
